dask/dataframe/_pyarrow.py

```diff
@@ -101,3 +104,21 @@ def check_pyarrow_string_supported():
             "Using dask's `dataframe.convert-string` configuration "
             "option requires `pandas>=2.0` to be installed."
         )
+
+
+def df_from_pyarrow_table(table, **kwargs):
```

Comment:
Update: I pulled the "metadata trick" out of this PR for now. We can revisit that idea if the p2p approach eventually needs to avoid carrying around the `meta` object.

Reply:
I don't see any reason to pass the metadata in any other place but `shuffle_unpack`, so I think this will stay.